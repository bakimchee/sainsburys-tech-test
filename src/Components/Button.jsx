import styled from 'styled-components';
import { theme as staticTheme } from '../Theme';

const Button = styled.button`
  padding: 0.5rem 1rem;
  color: white;
  background: ${staticTheme.color.palette.orange};
  font-weight: bold;
  border: none;
  border-radius: ${staticTheme.border.radius.default};
  text-align: center;
  text-decoration: none;
  touch-action: manipulation;
  min-height: ${staticTheme.size.default};
  min-width: ${staticTheme.size.default};
  display: flex;
  justify-content: center;
  align-items: center;
  &:focus,
  :hover {
    background-color: ${staticTheme.color.palette.orangeDark};
    cursor: pointer;
  }
  // https://github.com/philipwalton/flexbugs#9-some-html-elements-cant-be-flex-containers
  &:not(button) {
    align-items: center;
    display: inline-flex;
    justify-content: center;
  }
  ${(props) =>
    props.fullWidth &&
    `
    display: block;
    width: 100%;
  `}
  ${(props) =>
    props.outlined &&
    `
    background-color: transparent;
    border: solid ${staticTheme.border.width.large} ${staticTheme.color.palette.orange};
    color: ${staticTheme.color.palette.orange};
    &:focus,
    :hover {
      background-color: ${staticTheme.color.palette.orangeLight};
    }
  `}
  ${(props) =>
    props.small &&
    `
    padding: 0.25rem 0.5rem;
    font-size: small;
    `}
`;

export const TransparentButton = styled(Button)`
  background-color: transparent;
  color: ${({ theme }) => theme.text};
  &:focus,
  :hover {
    background-color: ${staticTheme.color.palette.orangeLight};
  }
`;

export const IconButton = styled(Button)`
  & * + * {
    margin-left: 0.5rem;
  }
`;

export default Button;
