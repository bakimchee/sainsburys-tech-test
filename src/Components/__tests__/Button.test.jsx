/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

test('Button', () => {
  const component = renderer.create(<Button>Add to basket</Button>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
