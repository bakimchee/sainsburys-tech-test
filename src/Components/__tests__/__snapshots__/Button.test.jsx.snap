// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  className="Button-uinp5t-0 hpjRxf"
>
  Add to basket
</button>
`;
